crates/dbsp/src/trace/mod.rs

```diff
@@ -489,9 +489,11 @@ where
     /// [`Batch::recede_to`].
     fn recede_to(&mut self, frontier: &Self::Time);
 
-    /// If this batch supports writing itself to storage, this causes it to
-    /// happen.
-    fn persist(&mut self) {}
+    /// If this batch is not on storage, but supports writing itself to storage,
+    /// this method writes it to storage and returns the stored version.
+    fn persisted(&self) -> Option<Self> {
```

Comment:
"persisted" suggests a getter to me
For the purposes described by the comment I think the previous name "persist" is a better match
Or `persist_if_possible` if you want to be pedantic
I would document the fact that this returns None if the batch is already persisted
(I initially expected it would return None if it cannot be persisted)